src/components/Header/Header.tsx

```diff
@@ -0,0 +1,80 @@
+import cn from 'classnames';
+import React, { useEffect, useRef } from 'react';
+
+import { ErrorType } from '../../types/ErorTypes';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  isAllCompleted: boolean;
+  onAdd: (todoTitle: string) => void;
+  onError: (error: ErrorType) => void;
+  onNewTodoTitle: (title: string) => void,
+  newTodoTitle: string
+  onUpdate: (id: number, todo: Todo) => void;
+  todos: Todo[]
+};
+
+export const Header: React.FC<Props> = ({
+  isAllCompleted,
+  onAdd,
+  onError,
+  onNewTodoTitle,
+  newTodoTitle,
+  onUpdate,
+  todos,
+}) => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleSubmitAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (newTodoTitle.trim()) {
+      onAdd(newTodoTitle);
+    } else {
+      onError(ErrorType.TitleError);
+    }
+  };
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [newTodoTitle]);
+
+  const handleUpdateTodoStatus = () => {
+    const newStatus = todos.some(todo => !todo.completed);
+
+    todos.forEach(todo => {
+      onUpdate(todo.id, {
+        ...todo,
+        completed: newStatus,
+      });
+    });
```

Comment:
If some of the todos are completed and a new status is completed, you don't need to send requests for them as they already have the same status.